scripts/main.ts

```diff
@@ -0,0 +1,128 @@
+"use strict";
+
+import { ServiceEndpointUIExtensionDetails } from 'ServiceEndpointDetails';
+import * as ko from 'knockout';
+
+var viewModel = new MyViewModel();
+function MyViewModel() {
+    this.isUpdate = ko.observable();
+    this.connectionName = ko.observable("");
+    this.scope = ko.observable("");
+    this.certificate = ko.observable("");
+    this.audience = ko.observable("");
+    this.issuer = ko.observable("");
+    this.projectid = ko.observable("");
+    this.privatekey = ko.observable("");
+    this.errors = ko.observable("");
+
+    this.connectionNameFieldColor = ko.computed(function () {
+        if (this.connectionName().length == 0)
```

Comment:
`this` is used inside a local function. Either use a lambda style `() => {`, or define an explicit `this` : `function (this: MyViewModel)`

Reply:
Fixed in next iteration